File: IBM-bank-backend/src/main/java/com/IBM/IBM_bank/Models/Cliente.java
package com.IBM.IBM_bank.Models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;

@Entity
public class Cliente {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false, length = 64)
    private String nome;

    @Column(nullable = false, length = 3)
    private Integer idade;

    @Column(nullable = false, length = 100)
    private String endereco;

    @Column(nullable = false, length = 50, unique = true)
    private String email;

    @JsonIgnore
    @OneToOne(mappedBy = "cliente", cascade = CascadeType.ALL)
    private Conta conta;

    // Getters e Setters

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public Integer getIdade() {
        return idade;
    }

    public void setIdade(Integer idade) {
        this.idade = idade;
    }

    public String getEndereco() {
        return endereco;
    }

    public void setEndereco(String endereco) {
        this.endereco = endereco;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public Conta getConta() {
        return conta;
    }

    public void setConta(Conta conta) {
        this.conta = conta;
    }
}

File: IBM-bank-frontend/src/shared/models/cliente.ts
export interface Cliente {
  id?:number, 
  nome: string, 
  idade: number,
  endereco: string,
  email: string, 
  numeroConta?: number,
}

export enum eColunaXPropCliente {
  // 'Nome da coluna' = 'nome da propriedade'
  'id' = 'id', 
  'Tipo' = 'tipo',
  'Nome' = 'nome', 
  'Idade' = 'idade',
  'Endereço' = 'endereco',
  'Email' = 'email', 
  'Número da conta' = 'numeroConta',
}